File: owlandfox.py
testcase = int(input())
def sum_of_number(number):
    number = str(number)
    sum = 0
    for value in number:
        sum = sum + int(value)
    return sum
def find_number(number, sum_required):
    while True:
        if sum_of_number(number) == sum_required:
            return number
        else:
            number = number - 1
for val in range(0,testcase):
    number = int(input())
    sum_required = sum_of_number(number) - 1
    if sum_required == 0:
        print("0")
    else:
        print(find_number(number,sum_required))

File: owlandfox.cpp
#include <iostream>
#include <unordered_set>
using namespace std;
int sum_of_number(int no){
    int sum =0;
    while(no>0)
    {
    int a=no%10;
    no=no/10;
    sum=sum+a;
    }
    return sum;
}
int find_number(int number, int sum){
    for (;;){
        if(sum_of_number(number) == sum){
            return number;
        }
        else{
            number--;
        }
    }
    return 0;
}
int main() 
{
    int test_case = 0;
    cin>>test_case;
    for (int i =0; i<test_case; i++){
        long int number;
        cin>>number;
        int sum = sum_of_number(number) - 1;
        if( sum == 0 ){
            cout<< "0"<<endl;
        }
        else{
        cout<<find_number(number , sum)<<endl;
        }
    }

    return 0;
}